tests/transactionScheduler_events_test.cdc

```diff
@@ -0,0 +1,261 @@
+    scheduleTransaction(
+        timestamp: timeInFuture,
+        fee: feeAmount,
+        effort: mediumEffort,
+        priority: mediumPriority,
+        data: testData,
+        testName: "Test Transaction Cancelation: First Scheduled",
+        failWithErr: nil
+    )
+
+    // Cancel the transaction
+    cancelTransaction(
+        id: transactionToCancel,
+        failWithErr: nil
+    )
+
+    let canceledEvents = Test.eventsOfType(Type<FlowTransactionScheduler.Canceled>())
+    Test.assert(canceledEvents.length == 1, message: "Should only have one Canceled event")
+    let canceledEvent = canceledEvents[0] as! FlowTransactionScheduler.Canceled
+    Test.assertEqual(transactionToCancel, canceledEvent.id)
+    Test.assertEqual(mediumPriority, canceledEvent.priority)
+    Test.assertEqual(feeAmount/UFix64(2.0), canceledEvent.feesReturned)
+    Test.assertEqual(feeAmount/UFix64(2.0), canceledEvent.feesDeducted)
+    Test.assertEqual(serviceAccount.address, canceledEvent.transactionHandlerOwner)
+    Test.assertEqual("A.0000000000000001.TestFlowScheduledTransactionHandler.Handler", canceledEvent.transactionHandlerTypeIdentifier!)
+
+    // Make sure the status is canceled
+    var status = getStatus(id: transactionToCancel)
+    Test.assertEqual(statusCanceled, status!)
+
+    // Available Effort should be completely unused
+    // for the slot that the canceled transaction was in
+    var effort = getSlotAvailableEffort(timestamp: timeInFuture, priority: mediumPriority)
+    Test.assertEqual(UInt64(mediumPriorityMaxEffort), effort!)
+
+    // Assert that the new balance reflects the refunds
+    Test.assertEqual(balanceBefore - feeAmount/UFix64(2.0), getBalance(account: serviceAccount.address))
+    Test.assertEqual(feesBalanceBefore + feeAmount/UFix64(2.0), getFeesBalance())
+}
+
+access(all) fun testScheduledTransactionExecution() {
+
+    var currentTime = getTimestamp()
+    var timeInFuture = currentTime + futureDelta
+
+    feesBalanceBefore = getFeesBalance()
+
+    var scheduledIDs = TestFlowScheduledTransactionHandler.scheduledTransactions.keys
+
+    // Simulate FVM process - should not yet process since timestamp is in the future
+    processTransactions()
+
+    // Check that no PendingExecution events were emitted yet (since transaction is in the future)
+    let pendingExecutionEventsBeforeTime = Test.eventsOfType(Type<FlowTransactionScheduler.PendingExecution>())
+    Test.assert(pendingExecutionEventsBeforeTime.length == 0, message: "PendingExecution before time")
+
+    // move time forward to trigger execution eligibility
+    // Have to subtract to handle the automatic timestamp drift
+    // so that the medium transaction that got scheduled doesn't get marked as pendingExecution
+    Test.moveTime(by: Fix64(futureDelta - 6.0))
+    while getTimestamp() < timeInFuture {
+        Test.moveTime(by: Fix64(1.0))
+    }
+
+    // Simulate FVM process - should process since timestamp is in the past
```

Comment:
it would be nice to add a comment which transaction it would process since we don't schedule anything in this test, I guess it should be something like:

```
// process transaction scheduled in the testTransactionScheduleEventAndData 
```